openprovider/openprovider.php

```diff
@@ -1312,14 +1324,26 @@ private function manageNameservers(
         $op_domain_request = $api->call('searchDomainRequest', $args);
         $this->logRequest($api);
 
-        if ($op_domain_request->getCode() != 0 || count($op_domain_request->getData()['results']) < 1) {
-            $this->Input->setErrors([
-                'errors' => [
-                    Language::_('OpenProvider.!error.domain.not_exist', true)
-                ]
-            ]);
+        $domain_request_failed = $op_domain_request->getCode() != 0;
 
-            return false;
+        if ($domain_request_failed) {
+            return $this->showError($op_domain_request->getMessage());
+        }
+
+        $domain_does_not_exist =
+            !isset($op_domain_request->getData()['results']) || // there is no results array in response data
+            empty($op_domain_request->getData()['results']); // or results array is empty
+
+        if (
+            $domain_does_not_exist &&
+            isset($service->status) &&
```

Comment:
if it's always true please remove it
else move this check to the top of the method if this code should run only if service active